packages/template-retail-react-app/app/components/store-locator-modal/index.jsx

```diff
@@ -5,89 +5,181 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import React, {useState} from 'react'
+import React, {useState, useRef, useEffect} from 'react'
 import PropTypes from 'prop-types'
 import {useIntl} from 'react-intl'
 import {
     Modal,
     ModalBody,
     ModalCloseButton,
     ModalContent,
-    useBreakpointValue
+    useBreakpointValue,
+    Button,
+    Box
 } from '@salesforce/retail-react-app/app/components/shared/ui'
 import StoreLocatorContent from '@salesforce/retail-react-app/app/components/store-locator-modal/store-locator-content'
 import {useSearchStores} from '@salesforce/commerce-sdk-react'
 import {
-    DEFAULT_STORE_LOCATOR_COUNTRY_CODE,
+    DEFAULT_STORE_LOCATOR_COUNTRY,
     DEFAULT_STORE_LOCATOR_POSTAL_CODE,
     STORE_LOCATOR_DISTANCE
 } from '@salesforce/retail-react-app/app/constants'
 import {noop} from '@salesforce/retail-react-app/app/utils/utils'
 import {useForm} from 'react-hook-form'
+import {isServer} from '@salesforce/retail-react-app/app/utils/utils'
 
-const StoreLocatorModal = ({isOpen, onClose = noop}) => {
+const StoreLocatorModal = ({onClose = noop}) => {
     const intl = useIntl()
+    var defaultSearchStoresParams = {
+        countryCode: DEFAULT_STORE_LOCATOR_COUNTRY.countryCode,
+        postalCode: DEFAULT_STORE_LOCATOR_POSTAL_CODE,
+        limit: 10
+    }
+    var defaultUserHasSetManualGeolocation = false
+    if (!isServer) {
+        var searchStoresParamsFromLocalStorage = JSON.parse(
+            window.localStorage.getItem('STORE_LOCATOR_SEARCH_STORES_PARAMS')
+        )
+        var userHasSetManualGeolocationFromLocalStorage = window.localStorage.getItem(
```

Comment:
The localstorage read operations and `JSON.parse()` happens in every re-render of this component. It could slow down the rendering of this component. Is it necessary to run these in every re-render?

Reply:
Is it absolutely necessary to store the state in localstorage? 

Generally speaking I think we want to avoid using localstorage because it's hard to maintain and making changes, for folks upgrading to different versions of the app, they'll often need to migrate the localstorage implementation and that's hard.

Reply:
the reading and writing of the store locator localstorage is scattered across different components, if we need these, we should consider extracting all the read/write logic into a single file.

Reply:
The states are stored in local storage so that when the modal, which is currently being completely loaded on or off conditionally, is closed and reopened the user input remains. 

Extracting the logic into one file sounds nice in most cases, but I also don't want to make more work for extensibility if it's better in the future that the logic is within each component that uses it, similar to my reason for not in [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1862#discussion_r1657704906).

Reply:
An alternative would be to maintain a Ref in `_app/index.js` and passing it into the store locator modal. Would that be more elegant than local storage?

Reply:
maintain a ref would make this component less portable...

How about wrapping the entire store locator modal in a react context

Reply:
I realized that as a result of our other conversation on making a "component layer", it's possible for me to pass a Ref into the `<YourModalThatFetchesThruHooks/>` component from the outermost StoreLocatorModal component to still have the overall StoreLocatorModal be portable. `_app/index.jsx` would still use `useDisclosure` to simply pass the props in and nothing else. A `useRef` or a `useContext` both would work

Reply:
The UI renders differently based on the values of the `userHasSetManualGeolocation` state, so in this case we should not use a Ref. A ref is typically used to hold references to a DOM element.

Reply:
Ah I see. Thank you. I thought bringing the relevant `useState`s that used to be stored in local storage in the child components to the higher component would be the simplest option, so I made this change. Since there is a higher level component, it can act as a component to persist states across closing and opening of the modal component.